### PR TITLE
Allow Travis to build tags and version branches

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Assumes tags are like `v1.0.0` and branches `1.0`.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Related doc: https://docs.travis-ci.com/user/customizing-the-build/#safelisting-or-blocklisting-branches.

**Comment by thewilkybarkid** [comment]
Thinking about it, we probably want to promote using forks, but if we do need local branches they should be like, say, `feature/foo` or `bug/bar`. In that case it'd be better to `except` anything containing a `/`?

**Comment by stephenwf** [comment]
Hmm, you can get some better control using [conditions in Travis](https://docs.travis-ci.com/user/conditions-v1). It's not clear from the doc, but I believe it a top level `if`:
```
if: tag IS present OR \
    type = pull_request OR \
    branch = master
```

You can also add these `if:` blocks to stages

**Comment by thewilkybarkid** [comment]
Ah, interesting. Still need to check if the branch name is `master` or version-like (eg `1.0`; or not prefixed).

**Comment by thewilkybarkid** [comment]
To add bit more context, want to tag `v0.1.0` and create a `0.1` branch (`master` then being `0.2`). Means can release a `v0.1.1` before `v0.2.0` if needed.

File: .travis.yml [hunk 1]
```diff
@@ -30,13 +30,20 @@ jobs:
...
 cache:
     directories:
         - $HOME/.composer/cache/files
 
-branches:
-    only:
-        - master
+if: |
+    branch = master OR \
+    branch =~ /^(?:[0-9]|[1-9][0-9]*)\.(?:[0-9]|[1-9][0-9]*)$/ OR \
```
Comment (giorgiosironi) [inline .travis.yml]: Version branches are probably a necessary evil for minor versions for releasing open source packages. The more complex this is the more we should start to extract it from the single repository into a template.

**Pull request merged**